sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1371 @@
+        return service.entitiesRecognitionGeneralWithRestResponseAsync(
+            batchInput,
+            options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of named entities input - {}", ignoredValue))
+            .doOnSuccess(response -> logger.info("A batch of named entities output - {}", response.getValue()))
+            .doOnError(error -> logger.warning("Failed to named entities - {}", error))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    // PII Entity
+    /**
+     * Returns a list of personal information entities ("SSN", "Bank Account", etc) in the text. For the list of
+     * supported entity types, check https://aka.ms/tanerpii. See https://aka.ms/talangs for the list of enabled
+     * languages.
+     *
+     * @param text the text to recognize PII entities for.
+     *
+     * @return A {@link Mono} containing the {@link RecognizeEntitiesResult PII entity} of the text.
+     *
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<RecognizePiiEntitiesResult> recognizePiiEntities(String text) {
+        try {
+            return recognizePiiEntitiesWithResponse(text, defaultLanguage).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns a list of personal information entities ("SSN", "Bank Account", etc) in the text. For the list of
+     * supported entity types, check https://aka.ms/tanerpii. See https://aka.ms/talangs for the list of enabled
+     * languages.
+     *
+     * @param text the text to recognize PII entities for.
+     * @param language The 2 letter ISO 639-1 representation of language for the text. If not set, uses "en" for
+     * English as default.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has the {@link
+     * RecognizeEntitiesResult named entity} of the text.
+     *
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RecognizePiiEntitiesResult>> recognizePiiEntitiesWithResponse(String text, String language) {
+        try {
+            return withContext(context -> recognizePiiEntitiesWithResponse(text, language, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<RecognizePiiEntitiesResult>> recognizePiiEntitiesWithResponse(String text, String language,
+        Context context) {
+        return recognizeBatchPiiEntitiesWithResponse(
+            Arrays.asList(new TextDocumentInput(Integer.toString(0), text, language)), null, context)
```

Comment:
Use Collections.singletonList rather than Arrays.asList because there is only one item in it.
It's easier to use a constant "0" instead of creating an object and then doing toString on it.

You don't need to use flatMap. .map is sufficient because there is nothing async about this. Then you don't have to allocate and wrap the response with Mono.just

Reply:
There are other instances of this. I won't point them all out. ex. L813